### PR TITLE
Add BypassPullRequestAllowances field

### PR DESCRIPTION
**Pull request by Starttoaster**
Closes https://github.com/google/go-github/issues/2431

### CODE REVIEW

File: github/repos.go [hunk 1]
```diff
@@ -1012,6 +1018,29 @@ type BranchRestrictionsRequest struct {
 	Apps []string `json:"apps,omitempty"`
 }
 
+// BypassPullRequestAllowances represents the people, teams, or apps who are allowed to bypass required pull requests. 
+type BypassPullRequestAllowances struct {
+	// The list of user slugs allowed to bypass pull request requirements.
```
Comment (gmlewis) [inline github/repos.go:1023]:
```suggestion
	// The list of users allowed to bypass pull request requirements.
```

File: github/repos.go [hunk 2]
```diff
@@ -1012,6 +1018,29 @@ type BranchRestrictionsRequest struct {
...
+	Users []*User `json:"users"`
+	// The list of team slugs allowed to bypass pull request requirements.
```
Comment (gmlewis) [inline github/repos.go:1025]:
```suggestion
	// The list of teams allowed to bypass pull request requirements.
```

File: github/repos.go [hunk 3]
```diff
@@ -1012,6 +1018,29 @@ type BranchRestrictionsRequest struct {
...
+	Teams []*Team `json:"teams"`
+	// The list of app slugs allowed to bypass pull request requirements.
```
Comment (gmlewis) [inline github/repos.go:1027]:
```suggestion
	// The list of apps allowed to bypass pull request requirements.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @Starttoaster !
Just a few minor changes, please.

**Comment by Starttoaster** [comment]
Thanks for the review @gmlewis ! Committed all suggestions.

**Review by gmlewis — APPROVED** [review]
Thank you, @Starttoaster !
LGTM.

Awaiting second LGTM from any other contributor to this repo before merging.

**Comment by Starttoaster** [comment]
> Thank you, @Starttoaster ! LGTM.
> 
> Awaiting second LGTM from any other contributor to this repo before merging.

Thanks!

**Comment by gmlewis** [comment]
Whups, I jumped the gun... it looks like some of your contributed files need to be run through `gofmt`, please.

**Comment by Starttoaster** [comment]
> Whups, I jumped the gun... it looks like some of your contributed files need to be run through `gofmt`, please.

Weird. Usually my text editor fmts for me, I guess it didn't because of the length of the file. Thanks!

**Comment by Starttoaster** [comment]
@gmlewis pardon me if this comes off as hasty, I just kind of want to switch my go.mods off of my fork, and open up this PR to see if it's merged once a day or so. I was curious what the usual expectation is for duration of time between first and second tier of PR reviews? Not trying to put pressure, I just want to set my expectations accordingly.

**Comment by gmlewis** [comment]
> @gmlewis pardon me if this comes off as hasty, I just kind of want to switch my go.mods off of my fork, and open up this PR to see if it's merged once a day or so. I was curious what the usual expectation is for duration of time between first and second tier of PR reviews? Not trying to put pressure, I just want to set my expectations accordingly.

Hi @Starttoaster - this repo is maintained by volunteers and we rely on other contributors to the repo to perform second code reviews when appropriate. Sometimes it is immediate, but more often it takes days or weeks. There are a few people that I can reach out to when there is some kind of urgency, but I prefer to not bother them unless there is. However, let's see if anyone steps up by tomorrow and if not, I can request that one of these trustworthy individuals take a look if they have time.

Thank you for your patience.

**Comment by Starttoaster** [comment]
It's not urgent or anything. No need to disrupt people. I can just keep my go.mods pinned to my fork for eternity if need be, it's just not really my preference is all. I was mostly just interested to hear how this process is intended to go. I enjoy the idea of community PR reviews, but in practice, it's been my observation that it's not usually a great experience for would-be contributors awaiting reviewers because most devs put their focus on their day jobs and then log out at the end of the day to spend time with their families (understandably, because that's how they make money to live.)

Anyway, tl;dr -- no need to move mountains for me. Just wanted to set my expectations correctly so I wouldn't feel compelled to load this PR page every morning.

**Review by raynigon — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @raynigon !
Merging.
